packages/viz/src/lib/Layer.ts

```diff
@@ -146,6 +150,110 @@ export class Layer {
   public get source() {
     return this._source;
   }
+
+  /**
+   * @public
+   * This method creates popups every time the
+   * user clicks on one or more features of the layer.
+   */
+  public async setPopupClick(elements: PopupElement[] | string[] | null = []) {
+    if (elements && elements.length > 0) {
```

Comment:
I think we should move this code to the Popup class class or abstract it in any way.

We are adding concepts that the layer shouldn't be aware of, and IMHO, we should create a Popup instance with methods to create handlers and call them when creating the layer.

Reply:
Not sure if I get you, do you mean Popup has a method to create a handler and pass it to the Layer constructor? Something like:

```javascript
const clickPopup = new carto.viz.Popup();
const hoverPopup = new carto.viz.Popup({closeButton: false});

const clickHandler = clickPopup.createHandler(['attr_name', 'attr_bar']);
const hoverHandler = hoverPopup.createHandler(['attr_name', 'attr_bar']);

const layer = new carto.viz.Layer('layer_name', { onClick: clickHandler, onHover: hoverHandler });
```

Reply:
Yes! That's what I kinda meant with my original comment!

Given that we don't probably want to let the user create those instances outside our Layer class (even though I see no problem in that), we can create those instances within our Layer class when using `setPopupClick` or `setPopupHover` but this way it will be much clearer when reading Layer's code.